### PR TITLE
ARROW-2855: [C++] Blog post that outlines the benefits of using jemalloc

### PR DESCRIPTION
**Pull request by xhochy**
I will update the date once this has passed review.

### CODE REVIEW

File: site/_posts/2018-07-01-jemalloc.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+applies to the C++/GLib/Python implementations of Arrow. In most cases changing
+the default allocator is normally done to avoid problems that occur with many
+small, frequent (de)allocations. In contrast, in Arrow we normally deal with
+large in-memory datasets. While jemalloc provides good strategies for [avoiding
+RAM fragmentation for allocations that are lower than a memory page
+(4kb)][1], it also provides functionality that improves performance on
+allocations that span several memory pages.
+
+Outside of Apache Arrow, [jemalloc powers the infrastructure of Facebook][2]
+(this is also where most of its development happens). t is also used as the
```
Comment (kou) [inline site/_posts/2018-07-01-jemalloc.md]: "t" -> "It"?

File: site/_posts/2018-07-01-jemalloc.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+64byte aligned. This is so that we can get the most performance out of SIMD
+instruction sets like AVX. While the most modern SIMD instructions also work on
+unaligned memory, their performance is much better on aligned memory. To get the
+best performance for our analytical applications, we want all memory to be
+allocated such that SIMD performance is maximized.
+
+For aligned allocations, the POSIX APIs only provide the
+`aligned_alloc(void** ptr, size_t alignment, size_t size)` function to
+allocate aligned memory. There is also 
+`posix_memalign( void **ptr, size_t alignment, size_t size)` to modify an
```
Comment (kou) [inline site/_posts/2018-07-01-jemalloc.md]: A white space in `( void` exists accidentally.

**Review by kou — APPROVED** [review]
+1
(I found some typos.)

File: site/_posts/2018-07-01-jemalloc.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
+---
+layout: post
+title: "Ligthning fast memory allocation in Apache Arrow with jemalloc"
```
Comment (wesm) [inline site/_posts/2018-07-01-jemalloc.md]: Typo.

Maybe "Faster, more scalable memory allocation..."? Trying to think of there's a better way to describe the benefits than "Lightning fast"

File: site/_posts/2018-07-01-jemalloc.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+With the release of the 0.9 version of Apache Arrow, we have switched our
+default allocator from the system allocator to jemalloc on OSX and Linux. This
```
Comment (wesm) [inline site/_posts/2018-07-01-jemalloc.md]: Say "default memory allocator" or if you want to be more specific "default memory allocator for array buffers" (though there are other kinds of buffers, these are the ones perhaps most of interest)

**Review by wesm — COMMENTED** [review]
Looks great, thank you for writing! I left a couple comments. Feel free to update date, merge, and publish to the Arrow website whenever you like; I can help promote on social media.

**Comment by pitrou** [comment]
Very informative! +1 from me.

**Comment by xhochy** [comment]
If there are no further comments, I will publish this tomorrow

**Comment by wesm** [comment]
+1 from me. Thanks @xhochy!

**Pull request closed without merging**